From 531ae7af75b2be2867814693f069fb51e3155341 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 12 Jun 2019 17:32:11 +0100
Subject: [PATCH] overlays: Update w1-gpio and w1-gpio-pullup

The parasitic power (power on data) feature is now enabled by
default in the w1-gpio driver, so update the README and make the
"pullup" parameter a no-op.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/README                     | 9 ++-------
 arch/arm/boot/dts/overlays/w1-gpio-overlay.dts        | 3 +--
 arch/arm/boot/dts/overlays/w1-gpio-pullup-overlay.dts | 3 +--
 3 files changed, 4 insertions(+), 11 deletions(-)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2261,9 +2261,7 @@ Info:   Configures the w1-gpio Onewire i
         Use this overlay if you *don't* need a GPIO to drive an external pullup.
 Load:   dtoverlay=w1-gpio,<param>=<val>
 Params: gpiopin                 GPIO for I/O (default "4")
-
-        pullup                  Non-zero, "on", or "y" to enable the parasitic
-                                power (2-wire, power-on-data) feature
+        pullup                  Now enabled by default (ignored)
 
 
 Name:   w1-gpio-pullup
@@ -2271,11 +2269,8 @@ Info:   Configures the w1-gpio Onewire i
         Use this overlay if you *do* need a GPIO to drive an external pullup.
 Load:   dtoverlay=w1-gpio-pullup,<param>=<val>
 Params: gpiopin                 GPIO for I/O (default "4")
-
-        pullup                  Non-zero, "on", or "y" to enable the parasitic
-                                power (2-wire, power-on-data) feature
-
         extpullup               GPIO for external pullup (default "5")
+        pullup                  Now enabled by default (ignored)
 
 
 Name:   wittypi
--- a/arch/arm/boot/dts/overlays/w1-gpio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/w1-gpio-overlay.dts
@@ -14,7 +14,6 @@
 				pinctrl-names = "default";
 				pinctrl-0 = <&w1_pins>;
 				gpios = <&gpio 4 0>;
-				rpi,parasitic-power = <0>;
 				status = "okay";
 			};
 		};
@@ -36,6 +35,6 @@
 				<&w1>,"reg:0",
 				<&w1_pins>,"brcm,pins:0",
 				<&w1_pins>,"reg:0";
-		pullup =        <&w1>,"rpi,parasitic-power:0";
+		pullup;		// Silently ignore unneeded parameter
 	};
 };
--- a/arch/arm/boot/dts/overlays/w1-gpio-pullup-overlay.dts
+++ b/arch/arm/boot/dts/overlays/w1-gpio-pullup-overlay.dts
@@ -14,7 +14,6 @@
 				pinctrl-names = "default";
 				pinctrl-0 = <&w1_pins>;
 				gpios = <&gpio 4 0>, <&gpio 5 1>;
-				rpi,parasitic-power = <0>;
 				status = "okay";
 			};
 		};
@@ -38,6 +37,6 @@
 				<&w1_pins>,"reg:0";
 		extpullup =     <&w1>,"gpios:16",
 				<&w1_pins>,"brcm,pins:4";
-		pullup =        <&w1>,"rpi,parasitic-power:0";
+		pullup;		// Silently ignore unneeded parameter
 	};
 };
